src/main.rs

```diff
@@ -4,44 +4,72 @@ use quicksilver::{
     Input, Result, Settings, Window,
 };
 
-use rand::distributions::{Distribution, Uniform};
+use rand::distributions::{Distribution, Uniform, WeightedIndex};
 use std::vec::Vec;
+use rand::prelude::ThreadRng;
 
+#[derive(Debug)]
 struct Node {
     x: f32,
     y: f32,
 }
 
+const HEIGHT: f32 = 1024.0;
+const LENGTH: f32 = 768.0;
+
+impl Node {
+    pub fn gen(mut rng: ThreadRng) -> Self {
+        let x_dist = Uniform::from(0.0..HEIGHT);
+        let y_dist = Uniform::from(0.0..LENGTH);
+        Node {
+            x: x_dist.sample(&mut rng),
+            y: y_dist.sample(&mut rng),
+        }
+    }
+}
+
 struct World {
     nodes: Vec<Node>,
     connections: Vec<(u32, u32)>,
 }
 
-fn gen_world() -> World {
-    let node_count = 20;
+impl World {
+    pub fn gen() -> World {
+        let node_count = 20;
 
-    let mut nodes = Vec::new();
+        let mut rng = rand::thread_rng();
+        let nodes: Vec<Node> = (0..node_count)
+            .map(|_| Node::gen(rng))
+            .collect();
 
-    let mut rng = rand::thread_rng();
-    let x_dist = Uniform::from(0.0..1024.0);
-    let y_dist = Uniform::from(0.0..768.0);
-    for _ in 0..node_count {
-        let new_node = Node {
-            x: x_dist.sample(&mut rng),
-            y: y_dist.sample(&mut rng),
+        let connections_count = 40;
+        let mut connections = Vec::new();
+        for _ in 0..connections_count {
+            let distrib = Uniform::from(0..nodes.len());
+            let node_index = distrib.sample(&mut rng);
+            let node = nodes.get(node_index).unwrap();
+            let weights: Vec<f32> = nodes.iter()
```

Comment:
you need to filter nodes:
- remove self node
- remove nodes that already have a connection to the current node

Reply:
or you can set probability for such nodes to 0.0